src/transforms/aws_ec2_metadata.rs

```diff
@@ -173,7 +174,14 @@ impl TransformConfig for Ec2Metadata {
             fields,
         );
 
-        client.refresh_metadata().await?;
+        // If initial metadata is not required, log and proceed. Otherwise return error.
+        if let Err(error) = client.refresh_metadata().await {
+            if self.required.unwrap_or(true) {
```

Comment:
Nit: I'd recommend moving the `unwrap_or` up to where the other defaults are set around line 162 to keep them all together.